tests/ui/manual_checked_sub.fixed

```diff
@@ -0,0 +1,144 @@
+#![allow(clippy::collapsible_if)]
+#![warn(clippy::manual_checked_sub)]
+
+fn positive_tests() {
+    let a: u32 = 10;
+    let b: u32 = 5;
+    let c = 3;
+
+    // Basic subtraction inside an if
+    if let Some(result) = a.checked_sub(b)  {
+        //~^ manual_checked_sub
+        let c = result;
+    }
+
+    // Basic subtraction inside an if with an else block
+    if let Some(result) = a.checked_sub(b)  {
+        //~^ manual_checked_sub
+        let c = result;
+    } else {
+        let c = a + b;
+    }
+
+    // Basic subtraction inside an if with an else-if block
+    if let Some(result) = a.checked_sub(b)  {
+        //~^ manual_checked_sub
+        let c = result;
+    } else if a < b {
+        let c = a + b;
+    }
+
+    // Decrementing inside an if condition
+    if let Some(result) = a.checked_sub(0)  {
+        //~^ manual_checked_sub
+        let _ = result;
+    }
+
+    // Variable subtraction used in a function call
+    if let Some(result) = a.checked_sub(b)  {
+        //~^ manual_checked_sub
+        some_function(result);
+    }
+
+    // Subtracting two suffixed literals
+    if let Some(result) = 10u32.checked_sub(5u32)  {
```

Comment:
I am confused by this one. If you are subtracting two literals, then you know that you can drop the whole `if` block, depending on whether the test succeeds or not, and you also know if the subtraction will work or not, so why suggest a checked sub?